src/runtime/virtcontainers/pkg/cgroups/cgroups.go

```diff
@@ -117,13 +118,32 @@ func sandboxDevices() []specs.LinuxDeviceCgroup {
 
 func NewCgroup(path string, resources *specs.LinuxResources) (*Cgroup, error) {
 	var err error
+	var cgroup cgroups.Cgroup
 
 	cgroupPath, err := ValidCgroupPath(path, IsSystemdCgroup(path))
 	if err != nil {
 		return nil, err
 	}
 
-	cgroup, err := cgroups.New(cgroups.V1, cgroups.StaticPath(cgroupPath), resources)
+	if IsSystemdCgroup(path) {
+		slice, unit, err := getSliceAndUnit(path)
+		if err != nil {
+			return nil, err
+		}
+		// github.com/containerd/cgroups doesn't support creating a scope unit with
+		// v1 cgroups against systemd, the following interacts directly with systemd
+		// to create the cgroup and then load it using containerd's api
+		err = createCgroupsSystemd(slice, unit, uint32(os.Getpid())) // adding runtime process, it makes calling setupCgroups redundant
```

Comment:
To my understanding we create the scope as per the container engine request and we will not create other cgroups there.
If we add the pid of the shim here, we will not honor sandbox_cgroup_only=false, right?
Maybe we could put a comment here and deal with this later.

Reply:
Yes, I left a TODO to address the  sandbox_cgroup_only=false case later, however, IIUC according to: https://github.com/kata-containers/kata-containers/blob/a7a47bd7d48b49c9cb207b93473f99fd7965c4e8/src/runtime/virtcontainers/sandbox.go#L2185-L2189
we add this process either way